mantidproject/mantid
# Export tick formatter commands to plot script

**Description of work.**
Commands for settings axis tick label formats are now generated by the plot script generator. They will be omitted if they are default values.

**To test:**
- Load a workspace and plot a spectrum.
- Double click an axis to bring up the "Edit axis" dialog.
- Change the "Tick Format" to "Scientfic Format".
- Click the script button and select "Script to clipboard".
- Close the plot.
- Paste the script in the Editor.
- You should see a line containing "set_major_formatter" for the axis that you changed the format for.
- Run the script and it should reproduce the plot, including the tick formatting.

The script commands should only appear for non-default formatters for each axis. There are unit tests that enforce this, but you can check by performing similar steps to above. The defaults are:
- Decimal format (ScalarFormatter) for a linear axis.
- Scientific format (LogFormatterSciNotation) for a log axis.

Fixes #31374 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
